logic/scorer/scorer.go

```diff
@@ -15,6 +15,23 @@ func CribPoints(lead model.Card, crib []model.Card) int {
 	return points(lead, crib, true)
 }
 
+const cardsPerHand = 5
+
+// map card value (1 to 13) to number of occurrences in the hand
+// we have to include index 0 and index 14 here because some calculations pass those values in
+type valueToCount [15]uint8
```

Comment:
:cactus: depending on the architecture, I wonder if this would be even more efficient with 16 uint8s instead of 15

Reply:
:mag: this uses more memory than is necessary. We can accomplish the same goal with only (at most) 10 bytes.

Reply:
```go
type cardCount struct{
  value uint8
  numInHand uint8
}
type cardsInHandCount [cardsPerHand]cardCount
```

and then we could use `cardsInHandCount` instead of `valueToCount`. This would reduce iteration time by 66% (from 15 elements to at most 5 elements).

Reply:
I don't think so; we need a way to lookup `numInHand` by `value`, and if we can't do it in constant time it doesn't help the iteration much :/

Reply:
Either that, or we sort by `value`; but sorting adds iteration as well. I'd rather pay in slightly more memory (a few bytes isn't going to hurt anyone) to save iteration. Unless there's something I'm missing?